src/main/java/ca/uhn/fhir/jpa/starter/CustomConsentService.java

```diff
@@ -0,0 +1,129 @@
+package ca.uhn.fhir.jpa.starter;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.stereotype.Service;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import ca.uhn.fhir.jpa.api.dao.DaoRegistry;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.interceptor.consent.ConsentOutcome;
+import ca.uhn.fhir.rest.server.interceptor.consent.IConsentContextServices;
+import ca.uhn.fhir.rest.server.interceptor.consent.IConsentService;
+
+@Service
+public class CustomConsentService implements IConsentService {
+
+  private static final String OAUTH_CLAIM_NAME = "patient";
+
+  private static final org.slf4j.Logger log =
+      org.slf4j.LoggerFactory.getLogger(CustomConsentService.class);
+
+  private OAuth2Helper oAuth2Helper = new OAuth2Helper();
+  DaoRegistry daoRegistry;
+
+  public CustomConsentService() {}
+
+  public CustomConsentService(DaoRegistry daoRegistry) {
+    this.daoRegistry = daoRegistry;
+  }
+
+  @Override
+  public ConsentOutcome startOperation(RequestDetails theRequestDetails,
+      IConsentContextServices theContextServices) {
+    if (oAuth2Helper.checkInPatientCompartment(theRequestDetails.getResourceName())) {
+      return ConsentOutcome.AUTHORIZED;
+    }
+    if (theRequestDetails.getHeader("Authorization") != null) {
+      String patientId = getPatientFromToken(theRequestDetails);
```

Comment:
Does any of the validation logic below matter if the `patientId` here is `null`?

Reply:
I have handled where patientId is null and allowed all the requests.